obspy/io/dmx/core.py

```diff
@@ -0,0 +1,137 @@
+"""
+INGV DMX bindings to ObsPy core module.
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+
+from tempfile import SpooledTemporaryFile
+from obspy.core.util.attribdict import AttribDict
+from obspy import Stream, Trace, UTCDateTime
+import numpy as np
+
+descript_trace_dtypes = np.dtype([('network', "4S"),
+                                 ("st_name", "5S"),
+                                 ("component", "1S"),
+                                 ("insstype", np.int16),
+                                 ("begintime", np.double),
+                                 ("localtime", np.int16),
+                                 ("datatype", "1S"),
+                                 ("descriptor", "1S"),
+                                 ("digi_by", np.int16),
+                                 ("processed", np.int16),
+                                 ("length", np.int32),
+                                 ("rate", np.float32),
+                                 ("mindata", np.float32),
+                                 ("maxdata", np.float32),
+                                 ("avenoise", np.float32),
+                                 ("numclip", np.int32),
+                                 ("timecorrect", np.double),
+                                 ("rate_correct", np.float32),
+                                  ])
+
+structtag_dtypes = np.dtype([("sinc", "1S"),
+                             ("machine", "1S"),
+                             ("id_struct", np.int16),
+                             ("len_struct", np.int32),
+                             ("len_data", np.int32)])
+
+types = {"s": ("uint16", "H", 2), "q": ("int16", "h", 2),
+         "u": ("uint16", "H", 2), "i": ("int16", "h", 2),
+         "2": ("int32", "i", 4), "l": ("int32", "i", 4),
+         "r": ("uint16", "H", 2), "f": ("float32", "f", 4),
+         "d": ("float64", "d", 8)}
+
+
+def readstructtag(fid):
+    y = AttribDict()
+    data = np.fromfile(fid, structtag_dtypes, 1)
+    for (key, (fmt, size)) in structtag_dtypes.fields.items():
+        if str(fmt).count("S") != 0:
+            y[key] = data[key][0].decode('UTF-8')
+        else:
+            y[key] = data[key][0]
+    return y
+
+
+def readdescripttrace(fid):
+    y = AttribDict()
+
+    data = np.fromfile(fid, descript_trace_dtypes, 1)
+
+    for (key, (fmt, size)) in descript_trace_dtypes.fields.items():
+        if str(fmt).count("S") != 0:
+            y[key] = data[key][0].decode('UTF-8')
+        else:
+            y[key] = data[key][0]
+
+    return y
+
+
+def readdata(fid, n, t):
+    target = types[t]
+    return np.fromfile(fid, eval("np.%s" % target[0]), n)
```

Comment:
No eval please.. one option could be `np.dtype(target[0]).type` I think or use importlib or `getattr(np, target[0])`

Reply:
done (first one)